Tests/BasicTests/TemporaryFileTests.swift

```diff
@@ -44,6 +44,33 @@ class TemporaryFileTests: XCTestCase {
         // File should be deleted now.
         XCTAssertFalse(isFile(filePath))
     }
+    
+    func testNoCleanupTemporaryFile() throws {
+        let filePath: AbsolutePath
+        do {
+            let file = try TemporaryFile(deleteOnClose: false)
+            
+            // Check if file is created.
+            XCTAssertTrue(isFile(file.path))
+            
+            // Try writing some data to the file.
+            let stream = BufferedOutputByteStream()
+            stream <<< "foo"
+            stream <<< "bar"
+            stream <<< "baz"
+            try fputs(stream.bytes.contents, file.fileHandle)
+            
+            // Go to the beginning of the file.
+            file.fileHandle.seek(toFileOffset: 0)
+            // Read the contents.
+            let contents = try? file.fileHandle.readFileContents()
+            XCTAssertEqual(contents, "foobarbaz")
```

Comment:
This test could be simplified I think. We can write data in the do block and then just read it outside the block using the `FileSystem` .

Reply:
Hmm I kept consistent with the other tests in this file, do you think they should all be changed?

Reply:
I think there was a reason to do that for other tests but anyway it was just a nitpick. LGTM!